packages/squiggle-lang/src/rescript/Distributions/SampleSetDist/SampleSetDist_ToPointSet.res

```diff
@@ -70,64 +55,53 @@ let toPointSetDist = (
     ~minDiscreteWeight=minDiscreteToKeep,
   )
 
-  let length = samples->E.A.length->float_of_int
+  let contLength = continuousPart->E.A.length
+  let length = samples->E.A.length
+  let outLength = contLength <= 5 ? length - contLength : length
+  let pointWeight = 1. /. outLength->float_of_int
+
+  let pdf = if contLength <= 5 {
+    // Drop these points; missing mass accounted for by outLength above
+    None
+  } else if E.A.unsafe_get(continuousPart, 0) == E.A.unsafe_get(continuousPart, contLength - 1) {
+    // All the same value: treat as discrete
+    Js.Array2.push(discretePart, (E.A.unsafe_get(continuousPart, 0), contLength))->ignore
+    None
```

Comment:
I think it's awkward that lines 63-69 are doing similar checks to 53-56. Maybe not worth doing now, but if we had a lot of time, I'd want these concerns to be better batched into one cluster; not two.